homeassistant/components/backup/manager.py

```diff
@@ -422,11 +430,21 @@ async def _async_upload_backup(
             return_exceptions=True,
         )
         for idx, result in enumerate(sync_backup_results):
+            if isinstance(result, BackupReaderWriterError):
+                # writer errors will affect all agents
+                # no point in continuing
+                raise BackupManagerError(str(result)) from result
+            if isinstance(result, BackupAgentError):
+                agent_errors[agent_ids[idx]] = result
+                continue
             if isinstance(result, Exception):
+                # trap bugs from agents
```

Comment:
Maybe we should elaborate in the comment we're not sure these errors are from the agent, but we treat them as such to not abort a backup receive or create operation?

Reply:
Why are we not sure it's from the agent?

Reply:
Can we guarantee the writer catches everything and re-raises?

Reply:
The only thing the writer does in `open_stream` that agents use from the writer, is open a file, read it, yield the chunks, and close it. We catch OSError around all that and reraise. I don't see what can error outside OSError for that.

Reply:
For the Supervisor writer it's one client call that is done and we catch `SupervisorError` around that and reraise.

Reply:
I think it's more about that we don't control all agents but we control the manager and writers. We can always fix bugs in the manager and writer but it's not certain that we can fix bugs in agents so we need the system to be resilient to agent bugs so that one bad agent doesn't take down the whole backup flow.